src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -118,7 +118,7 @@ public function validate($value, Constraint $constraint)
      */
     private function checkMX($host)
     {
-        return checkdnsrr($host, 'MX');
+        return !empty($host) && checkdnsrr($host, 'MX');
```

Comment:
Please avoid `empty` :) We expect a string, and only a empty string (`$host === ''`)  should not be passed to `checkdnsrr`.